undertow/WFLY-10084_Proxy_Protocol.adoc

```diff
@@ -0,0 +1,51 @@
+= WFLY-10084 Support for the PROXY protocol
+:author:            Stuart Douglas
+:email:             sdouglas@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org
+
+== Overview
+
+This feature is to add support for version 1 of the proxy protcol as defined at
+http://www.haproxy.org/download/1.8/doc/proxy-protocol.txt. This will be supported via a new attribute on Undertow
+listeners called `proxy-protocol`.
+
+== Issue Metadata
+
+=== Issue:
+
+* {issue-base-url}/WFLY-10084[WFLY-10084]
+
+=== Related Issues:
+
+* {issue-base-url}/UNDERTOW-573[UNDERTOW-573]
+* {issue-base-url}/EAP7-974[EAP7-974]
+
+=== Dev Contacts:
+
+* mailto:{email}[{author}]
+
+=== QE Contacts:
+
+=== Affected Projects or Components:
+
+* Undertow
+
+== Requirements
+
+This will be implemented as an attribute called `proxy-protocol` under each of the listeners. If this is set to true
+then the listener will be wrapped by the Undertow proxy protocol implementation. This wrapper reads the proxy protocol
+header from the stream, before handing off the connection to the normal listener implementation.
```

Comment:
What does it do with the information?  IOW how is supporting this protocol going to help solve concrete problems?  The ones I've heard about relate to making the original client address available to things like access loggers and calls to Keycloak.

I think I know in a vague hand-wavy kind of way. :)

Reply:
Ditto @bstansberry's question, the driving requirement is from access loggers, keycloak, etc.